sagemaker-featurestore/feature_store_securely_store_images.ipynb

```diff
@@ -0,0 +1,680 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "time.sleep(30)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s3_client = sagemaker_session.boto_session.client('s3', region_name=region)\n",
+    "\n",
+    "feature_group_s3_uri = feature_group.describe().get(\"OfflineStoreConfig\").get(\"S3StorageConfig\").get(\"ResolvedOutputS3Uri\")\n",
+    "\n",
+    "feature_group_s3_prefix = feature_group_s3_uri.replace(f\"s3://{s3_bucket_name}/\", \"\")\n",
+    "offline_store_contents = None\n",
+    "while (offline_store_contents is None):\n",
+    "    objects_in_bucket = s3_client.list_objects(Bucket=s3_bucket_name,Prefix=feature_group_s3_prefix)\n",
+    "    if ('Contents' in objects_in_bucket and len(objects_in_bucket['Contents']) > 1):\n",
+    "        offline_store_contents = objects_in_bucket['Contents']\n",
+    "    else:\n",
+    "        print('Waiting for data in offline store...\\n')\n",
+    "        time.sleep(60)\n",
+    "    \n",
+    "print('Data available.')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Use Amazon Athena to Query your Encrypted Data in your Feature Store\n",
+    "Using Amazon Athena, we query the image data set that we stored in our feature store to demonstrate how to extract your data set of images."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "query = feature_group.athena_query()\n",
+    "\n",
+    "table = query.table_name\n",
+    "\n",
+    "query_table = 'SELECT * FROM \"'+table+'\"'\n",
+    "\n",
+    "print('Running ' + query_table)\n",
+    "\n",
+    "# Run the Athena query\n",
+    "query.run(query_string=query_table, output_location='s3://'+s3_bucket_name+'/'+prefix+'/query_results/')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "time.sleep(60)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = query.as_dataframe()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(dataset.dtypes)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Below is the data queried from your feature store."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def parse_show_image(df):\n",
+    "    \"\"\"\n",
+    "    df: dataframe of your data\n",
+    "    \n",
+    "    Return a numpy array of your images that have been reshaped into it's corresponding shape. \n",
```

Comment:
"    Return a numpy array of your images that have been reshaped into a corresponding shape. \n",